### ISSUE TITLE
AssertionError assert row.max().item() < sparse_sizes[0] after updating

### ISSUE THREAD

**Issue by mariolinovIC** [rusty1s/pytorch_geometric#2913]
Hi, 

I was using the function torch_scatter.coalesce before with no problems. I updated today to the lasted version of pytorch geometric available via conda and now I get this error message:
`
assert row.max().item() < sparse_sizes[0]
AssertionError
`
I have checked the torch_sparse.coalesce history and it seems this part of the code has been modified recently [https://github.com/rusty1s/pytorch_sparse/blob/8c25ddefdc9f6e797aafff5782d1840025f80fa2/torch_sparse/storage.py#L21](url). 
Is there a bug in the latest version installed via conda ? Or it must be something I am doing wrong?
Thanks

**Comment by mariolinovIC** [comment]
Hi again, 

I modified the indexing I use for the cluster tensor passed as input to torch_sparse.coalesce and it works again. I am not sure that such assert is needed in torch_sparse.coalesce though.

Having a look to pool_edge in torch_geometric/nn/pool/pool.py [(https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/pool/pool.py)](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/pool/pool.py) and torch_sparse.coalesce I noticed that I don't understand quite well what should be the `n, m` input arguments to torch_sparse.coalesce.
For example in pool_edge `n, m = num_nodes, num_nodes ` before clustering, my thought is that they should be equal to the number of nodes after clustering.
Thank you for any help in understanding this.

Best regards, 

Mario

**Comment by rusty1s** [comment]
The assertion shouldn't have any effect as long as the `SparseTensor` is created correctly, i.e., the indices of non-zero values should never exceed the actual sparse matrix dimensions (which results in undefined behavior).

The `coalesce` call just orders the indices in a sparse matrix and removes duplicated entries via aggregation. In particular, `n` and `m` describe the dimensions of your sparse matrix.

**Comment by mariolinovIC** [comment]
Thank you for your answer. Then I think pool_edge in  torch_geometric/nn/pool/pool.py ([https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/pool/pool.py](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/pool/pool.py) ) should have n and m in coalesce equal to `cluster.max().item()+1`, not `cluster.size(0)`. Isnt it?

**Comment by rusty1s** [comment]
That is correct (although the output will always be correct anyway since `cluster.size(0) > cluster.max()`. Do you wanna fix it anyway?

**Comment by mariolinovIC** [comment]
I dont have problems with pool_edge, but I have a similar function to it. And in my case I had problems due to the indexing I was using for the nodes did not meet the assert condition. Altough I think that for a standard use it should not cause issues as you pointed. So no worries.
